### PR TITLE
feat(flags): add ApplicationFlags.application_auto_moderation_rule_create

### PR DESCRIPTION
**Pull request by spifory**
## Summary

Implements discord/discord-api-docs#6021

Note that I am unsure if the name of the flag is okay, I've just used what it was called in the documentation.

## Checklist

- [x] If code changes were made then they have been tested.
  - [x] I have updated the documentation to reflect the changes.
  - [ ] I have run `task pyright` and fixed the relevant issues.
- [ ] This PR fixes an issue.
- [x] This PR adds something new (e.g. new method or parameters).
- [ ] This PR is a breaking change (e.g. methods or parameters removed/renamed)
- [ ] This PR is **not** a code change (e.g. documentation, README, ...)

### CODE REVIEW

**Comment by toolifelesstocode** [comment]
The client does say that, however, remember that some names on the client are different on the API. Such as how there is 'guild' and 'server'

**Review by Skelmis — APPROVED** [review]
Looks good to me, i also agree with the naming as it keeps it consistent. All other values in the class mirror the documentation 1-1 so while this can be considered long it would also be consistent with both the class and discord docs

File: nextcord/flags.py [hunk 1]
```diff
@@ -1193,6 +1193,11 @@ class ApplicationFlags(BaseFlags):
         rather than using this raw value.
     """
 
+    @flag_value
+    def application_auto_moderation_rule_create(self):
+        """:class:`bool`: Returns ``True`` if the application uses the Auto Moderation API."""
```
Comment (ooliver1) [inline nextcord/flags.py:1198]:
```suggestion
        """:class:`bool`: Returns ``True`` if the application uses the Auto Moderation API.

        .. versionadded:: 2.5
        """
```

**Review by ooliver1 — CHANGES_REQUESTED** [review]

File: nextcord/flags.py [hunk 2]
```diff
@@ -1193,6 +1193,14 @@ class ApplicationFlags(BaseFlags):
         rather than using this raw value.
     """
 
+    @flag_value
+    def application_auto_moderation_rule_create(self):
+        """:class:`bool`: Returns ``True`` if the application uses the Auto Moderation API.
+
+        .. versionadded:: 2.5
```
Comment (EmreTech) [inline nextcord/flags.py:1200]:
```suggestion
        .. versionadded:: 2.6
```

**Review by EmreTech — APPROVED** [review]

**Pull request merged**